import "common.proto";
import "player.proto";

// 游戏服登录 command：4
message GameLoginRequest{
    required string token = 1;
    optional int32 plat_id = 2;  // ios 0 android 1
    optional string client_version = 3;  // 客户端版本
    optional string system_software = 4;  // 移动终端操作系统
    optional string system_hardware = 5;  // 移动终端机型
    optional string telecom_oper = 6;  // 运营商
    optional string network = 7;  // 3G WIFI 2G 4G
    optional int32 screen_width = 8;  // 显示屏宽度
    optional int32 screen_hight = 9;  // 显示屏高度
    optional float density = 10;  // 像素密度
    optional string login_channel = 11;  //登录渠道
    optional string mac = 12;  // MAC地址
    optional string cpu_hardware = 13;  // cpu类型 频率 核数
    optional int32 memory = 14;  // 内存信息单位M
    optional string gl_render = 15;  // opengl render 信息
    optional string gl_version = 16;  // opengl 版本信息
    optional string device_id = 17;  // 设备ID
    optional int32 platform = 18; //平台
    optional string open_id = 19;//
    optional string open_key = 20;//
    optional string pay_token = 21;//
    optional string appid = 22;//
    optional string appkey = 26;//
    optional string pf = 23;//
    optional string pfkey = 24;//
    optional string zoneid = 25;//
    optional int32 recharge_id = 27;// recharge_config 用于充值
}

//返回结果
message GameLoginResponse{
    required CommonResponse res = 1;
    optional int32 id = 2;
    optional string nickname = 3;
    //等级
    optional int32 level = 4;
    optional int32 exp = 5;
    //金币
	repeated int32 finances = 6;
    //抽取时间
    optional int32 fine_hero = 7;
    optional int32 excellent_hero = 8;
    optional int32 fine_equipment = 9;
    optional int32 excellent_equipment = 10;
    //other
    optional int32 pvp_times = 11;
    optional int32 pvp_refresh_count = 12;
    optional int32 vip_level = 13;
    optional int32 server_time = 14;
    optional int32 guild_id = 15;
    optional float combat_power = 16;
	optional int32 newbee_guide_id = 17;
	optional int32 register_time = 24;
    
    //stamina
    optional int32 get_stamina_times = 18; //通过邮件获取体力次数
    optional int32 buy_stamina_times = 19; //购买体力次数
    optional int32 last_gain_stamina_time = 20; //上次获得体力时间
    optional int32 soul_shop_refresh_times = 21; //武魂商店刷新次数刷新次数
	repeated int32 head = 22;
	optional int32 now_head = 23;
	repeated int32 first_recharge_ids = 25;
	optional int32 gag = 26;
	optional int32 closure = 27;
	optional int32 recharge = 28; // 累积充值
	optional int32 tomorrow_gift = 29; //
	optional int32 battle_speed = 30; //
    //抽取次数
    optional int32 fine_hero_times = 31;// 良将抽取次数
    optional int32 excellent_hero_times = 32; //神将抽取次数
    optional int32 fight_power_rank = 33; //战力排行
    optional int32 pvp_overcome_index = 34;
    optional int32 pvp_overcome_stars = 36;
    optional int32 pvp_overcome_refresh_count = 37;
    //
    repeated BuyStaminaTimes buy_times = 38; //购买资源（体力，讨伐令，鞋子）次数
    //
    optional bool is_open_next_day_activity = 39; // 是否开启次日功能
    //
    optional int32 first_recharge_activity = 40; // 是否已获得首次充值 -1:没领过 1:领过 
    optional float hight_power = 41; // 最高战力
	optional int32 story_id = 42; // 关卡剧情
    repeated int32 button_one_time = 43; //  0. 第二天活动开启后的按钮 1. 首次充值奖励 2. 关卡点我有惊喜
    optional int32 server_open_time = 44; //  开服时间
    optional string q360_recharge_url = 45;
  optional string one_dollar_flowid = 46;

}

message BuyStaminaTimes
{
    optional int32 resource_type = 1; //资源类型
    optional int32 buy_stamina_times = 2; //购买次数
    optional int32 last_gain_stamina_time = 3; //上次获得体力时间
}

//心跳响应88
message HeartBeatResponse{
    optional int32 server_time = 1;
}

//可恢复性资源，体力, 讨伐令等 2201, 2202
message StaminaOperRequest{
    required FinanceChanges finance_changes = 2;
}
//返回 2201 2202 
message StaminaOperResponse{
    required CommonResponse res = 1;
    required FinanceChanges finance_changes = 2;
    optional int32 buy_times = 3;
}
